app/jobs/sitemap_monitor_job.rb

```diff
@@ -1,9 +1,10 @@
+# frozen_string_literal: true
+
 class SitemapMonitorJob < ApplicationJob
   queue_as :sitemap
 
   def perform
-    SearchgovDomain.ok.each do |searchgov_domain|
-      searchgov_domain.index_sitemaps
-    end
+    SearchgovDomain.not_ok.each(&:check_status)
+    SearchgovDomain.ok.each(&:index_sitemaps)
```

Comment:
Please use `find_each` for these, rather than `each`. That will process the `SearchgovDomain` records in batches, rather than loading all records within the scope into memory. There are only several hundred records in that table at the moment, but this will protect us as that table grows.